source/tool/chipsec/hal/spi.py

```diff
@@ -525,6 +525,13 @@ def _send_spi_cycle(self, hsfctl_spi_cycle_cmd, dbc, spi_fla ):
 
         return cycle_done
 
+    def check_hardware_sequencing(self):
+        # Test if the flash decriptor is valid (and hardware sequencing enabled)
+        hsfsts = self.spi_reg_read( self.hsfs_off, 2 )
+        if not (hsfsts & Cfg.PCH_RCBA_SPI_HSFSTS_FDV):
+            logger().error("HSFS.FDV is 0, hardware sequencing is disabled")

```

Comment:
Please change to use read_register[_field] instead to make the code platform independent:

fdv = chipsec.chipset.read_register_field( self.cs, 'HSFS', 'FDV' )
if fdv == 0:
    ... raise exception